### PR TITLE
Rework comparison with other frameworks

### PR DESCRIPTION
**Pull request by KnorpelSenf**
As promised in https://t.me/grammyjs/244931

### CODE REVIEW

File: site/docs/resources/comparison.md [hunk 1]
```diff
@@ -171,16 +127,36 @@ People have different taste, so take this section with a grain of salt.
...
+### Frameworks Written in Go
+
+If you are proficient in both TypeScript and Go, then a reasonable metric for deciding on a language for your bot is the balance between development speed and execution speed.
+
+Pick grammY if you are not completely sure what you are building.
+TypeScript lets you iterate on your code base at incredible speeds.
+It is great for rapid prototyping, trying out new things, getting to know bots, and getting things done quickly.
+As a rule of thumb, processing ~100,000,000 updates per day can be done easily with TypeScript, but going beyond that will require extra work, such as using one more grammY plugin.
+
+Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you alreay know that your bot will process very high numbers of updates.
```
Comment (roj1512) [inline site/docs/resources/comparison.md:148]:
```suggestion
Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process very high numbers of updates.
```

File: site/docs/resources/comparison.md [hunk 2]
```diff
@@ -171,16 +127,36 @@ People have different taste, so take this section with a grain of salt.
...
+### Frameworks Written in Go
+
+If you are proficient in both TypeScript and Go, then a reasonable metric for deciding on a language for your bot is the balance between development speed and execution speed.
+
+Pick grammY if you are not completely sure what you are building.
+TypeScript lets you iterate on your code base at incredible speeds.
+It is great for rapid prototyping, trying out new things, getting to know bots, and getting things done quickly.
+As a rule of thumb, processing ~100,000,000 updates per day can be done easily with TypeScript, but going beyond that will require extra work, such as using one more grammY plugin.
+
+Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process very high numbers of updates.
```
Comment (KnorpelSenf) [inline site/docs/resources/comparison.md:148]:
```suggestion
Pick a library written in Go if you already know fairly well what you will be building (you don't expect to need much assistance), and you already know that your bot will process a very large number of updates.
```

Comment (KnorpelSenf) [reply site/docs/resources/comparison.md:148]: @roj1512 accept this if you think that it is an improvement to the wording

File: site/docs/resources/comparison.md [hunk 3]
```diff
@@ -171,16 +127,36 @@ People have different taste, so take this section with a grain of salt.
...
+This is much less relevant when you write a bot because most of the time is spent waiting for the network, but eventually, it will start to matter how fast your bot can parse JSON.
+Go can be a better choice in these cases.
+
+### Frameworks Written in Rust
+
+A similar point can be made [as with Go](#frameworks-written-in-go), but it is even stronger with Rust.
+In a way, it will take you even more time to write Rust, but your bot will be even faster, too.
+
+Also, please not that using Rust is fun but rarely necessary for bots.
+If you want to use Rust, then do it, but consider saying that you love Rust and not that it is the right tool for the job.
```
Comment (niusia-ua) [inline site/docs/resources/comparison.md:159]: `please not that`?

Is this correct?

Comment (KnorpelSenf) [reply site/docs/resources/comparison.md:159]: No. It should be _please note that_

Comment (niusia-ua) [reply site/docs/resources/comparison.md:159]:
```suggestion
Also, please note that using Rust is fun but rarely necessary for bots.
If you want to use Rust, then do it, but consider saying that you love Rust and note that it is the right tool for the job.
```

Comment (KnorpelSenf) [reply site/docs/resources/comparison.md:159]: Just here: 
```suggestion
Also, please note that using Rust is fun but rarely necessary for bots.
If you want to use Rust, then do it, but consider saying that you love Rust and not that it is the right tool for the job.
```

File: site/docs/uk/resources/comparison.md [hunk 4]
```diff
@@ -141,22 +87,32 @@ grammY і Telegraf мають різні кодові бази.
...
+
+Ви щойно створили нову чудову бібліотеку, а ми про неї ще не знаємо?
 Ви можете змінювати цю сторінку і додавати порівняння або поділитися з нами своїми думками в [груповому чаті](https://t.me/grammyjs)!
 
 ## Порівняння з фреймворками з інших мов програмування
 
 Існують причини надавати перевагу іншій мові програмування замість TypeScript.
 Найважливіше, щоб вам подобалося працювати з вашими інструментами та мовами.
-Якщо ви твердо налаштовані використовувати щось інше, то можете припинити читання тут.
+Якщо ви твердо налаштовані використовувати щось іншу мову програмування, то можете припинити читання тут.
```
Comment (LWJerri) [inline site/docs/uk/resources/comparison.md:100]:
```suggestion
Якщо ви твердо налаштовані використовувати іншу мову програмування, то можете припинити читання тут.
```

File: site/docs/uk/resources/comparison.md [hunk 5]
```diff
@@ -141,22 +87,32 @@ grammY і Telegraf мають різні кодові бази.
...
 
 Оскільки ви продовжуєте читати, напевно, вам цікаво, чому grammY написаний мовою TypeScript і чому вам теж варто розглянути можливість вибору цієї мови для вашого бота.
 
 У цьому розділі ми розглянемо, якими є переваги TypeScript порівняно з іншими мовами, коли мова йде про розробку Telegram ботів.
-Іншою поширеною мовою для розробки чат-ботів для Telegram є Python, тому ми зупинимося саме на ній.
-Деякі з наведених нижче тез є скоріше особистими думками, ніж обʼєктивними фактами.
+Це порівняння буде обмежено Python, Go та Rust.
```
Comment (LWJerri) [inline site/docs/uk/resources/comparison.md:105]:
```suggestion
Це порівняння буде обмежене Python, Go та Rust.
```

**Review by LWJerri — APPROVED** [review]
not :shipit:, bc I leave some comments.

File: site/docs/uk/resources/comparison.md [hunk 6]
```diff
@@ -47,82 +44,31 @@ Telegraf дивовижна бібліотека, і без неї grammY не 
...
-Основна перевага grammY над Telegraf 4 полягає в тому, що **він набагато простіший**.
-Наприклад:
+Ось список причин, чому варто використовувати grammY замість Telegraf:
 
+- grammY завжди підтримує останню версію Bot API.
+  Telegraf часто відстає на кілька версій.
 - grammY має [документацію](../).
-  Telegraf її не має: замість неї згенерований довідник API, який не містить пояснень.
-- Типи в grammY _просто працюють_ і будуть відповідати твоєму коду.
+  Telegraf її не має: замість неї слугує згенерований довідник API, який не містить пояснень, а ті кілька посібників, що існують, є неповними і їх важко знайти.
```
[the remark this answers is not in the document]
Comment (niusia-ua) [reply site/docs/uk/resources/comparison.md:57]: Yes. Telegraf doesn't have the documentation. "Telegraf (кого/що?) її (документації) не має".

Comment (niusia-ua) [inline site/docs/uk/resources/comparison.md:57]:
```suggestion
  Telegraf не має її: замість неї слугує згенерований довідник API, який не містить пояснень, а ті кілька посібників, що існують, є неповними і їх важко знайти.
```

**Review by niusia-ua — APPROVED** [review]

File: site/docs/uk/resources/comparison.md [hunk 7]
```diff
@@ -47,82 +44,31 @@ Telegraf дивовижна бібліотека, і без неї grammY не 
...
+- grammY використовує TypeScript; типи _просто працюють_ і будуть відповідати вашому коду.
   У Telegraf часто потрібно писати код певним чином, інакше він не скомпілюється --- попри те, що насправді він працюватиме без проблем.
 - grammY інтегрує підказки з [офіційного довідника Bot API](https://core.telegram.org/bots/api), які допомагають вам під час написання коду.
   Telegraf не надає вам жодних пояснень щодо вашого коду.
+- Багато інших речей, як-от краща продуктивність, велика екосистема плагінів, документація, перекладена для мільярдів людей, краща інтеграція з базами даних і вебфреймворками, краща сумісність під час виконання, розширення [VS Code](https://marketplace.visualstudio.com/items?itemName=grammyjs.grammyjs) тощо, які ви відкриєте для себе в процесі роботи.
```
Comment (deptyped) [inline site/docs/uk/resources/comparison.md:62]:
```suggestion
- Багато інших речей, як-от краща продуктивність, велика екосистема плагінів, документація, перекладена для мільярдів людей, краща інтеграція з базами даних і вебфреймворками, краща сумісність з середовищами виконання, розширення [VS Code](https://marketplace.visualstudio.com/items?itemName=grammyjs.grammyjs) тощо, які ви відкриєте для себе в процесі роботи.
```

File: site/docs/uk/resources/comparison.md [hunk 8]
```diff
@@ -171,16 +127,36 @@ grammY і Telegraf мають різні кодові бази.
...
+Якщо ви володієте і TypeScript, і Go, то розумною мірою для вибору мови для вашого бота є баланс між швидкістю розробки та швидкістю виконання.
+
+Обирайте grammY, якщо ви не зовсім впевнені в тому, що створюєте.
+TypeScript дозволяє ітераційно змінювати кодову базу з неймовірною швидкістю.
+Він чудово підходить для швидкого створення прототипів, випробовування нових речей, знайомства з ботами та швидкого виконання завдань.
+Як правило, за допомогою TypeScript можна легко обробляти ~100 000 000 оновлень на день, але якщо вийти за рамки цього показника, то знадобиться додаткова робота, наприклад, використання ще одного плагіна grammY.
+
+Вибирайте бібліотеку, написану на Go, якщо ви вже досить добре знаєте, що будете створювати (ви не очікуєте, що вам знадобиться велика допомога), і ви вже знаєте, що ваш бот буде обробляти дуже велику кількість оновлень.
+Як мова, що компілюється нативно, Go перевершує TypeScript за швидкістю роботи процесора на кілька порядків.
+Це набагато менш актуально, коли ви пишете бота, тому що більша частина часу витрачається на очікування мережі, але з часом це почне мати значення, як швидко ваш бот може розбирати JSON.
```
Comment (deptyped) [inline site/docs/uk/resources/comparison.md:150]:
```suggestion
Це набагато менш актуально, коли ви пишете бота, тому що більша частина часу витрачається на очікування мережі, але з часом почне мати значення, як швидко ваш бот може розбирати JSON.
```

File: site/docs/uk/resources/comparison.md [hunk 9]
```diff
@@ -171,16 +127,36 @@ grammY і Telegraf мають різні кодові бази.
...
+У таких випадках Go може бути кращим вибором.
+
+### Фреймворки, написані на Rust
+
+Можна зробити схоже зауваження, [як і у випадку з Go](#фреймворки-написані-на-go), але у випадку з Rust воно ще сильніше.
+У певному сенсі, вам знадобиться навіть більше часу на написання Rust, але ваш бот буде ще швидшим.
+
+Також, будь ласка, зверніть увагу, що використання Rust є цікавим, але рідко необхідним для ботів.
+Якщо ви хочете використовувати Rust, то робіть це, але не забувайте наголошувати на тому, що ви любите Rust і зазначати, що це правильний інструмент для роботи.
```
Comment (deptyped) [inline site/docs/uk/resources/comparison.md:159]:
```suggestion
Якщо ви хочете використовувати Rust, то робіть це, але слід враховувати, що це скоріше ваша любов до Rust, ніж те, що він є правильним інструментом для цієї роботи.
```

**Review by deptyped — CHANGES_REQUESTED** [review]

File: site/docs/resources/comparison.md [hunk 10]
```diff
@@ -47,82 +44,31 @@ This allowed to skip a lot of the frustrating discussions around how to cope wit
...
 
-##### Advantages of Telegraf
-
-- Older, thus more mature
-- Many more example bots, both in the repository and in the wild
+- You already have a large bot written in Telegraf and you no longer really work on it.
+  In that case, migrating to grammY may take more time than you will save in the long run, no matter how smooth the migration is.
+- You know Telegraf like the back of your hand and you do not care about changing your skill set.
+  grammY introduces a number of novel concepts that can be unfamiliar if you have used Telegraf only, and using grammY means that you will be exposed to new things.
+- There are a few details where Telegraf and grammY use different syntax to achieve the same thing, and you just happen to prefer one style of the other.
```
Comment (roziscoding) [inline site/docs/resources/comparison.md:70]:
```suggestion
- There are a few details where Telegraf and grammY use different syntax to achieve the same thing, and you just happen to prefer one style over the other.
```

File: site/docs/resources/comparison.md [hunk 11]
```diff
@@ -141,22 +87,32 @@ You don't want that.
...
 
 Given that you're still reading, you may want to know why grammY is written in TypeScript, and why you should maybe consider picking this language for your bot, too.
 
 This section will outline how TypeScript has a few advantages over other languages when it comes to developing Telegram bots.
-The main other language that is used to develop chat bots for Telegram is Python, so we will limit ourselves to this for now.
-Some of the following points are rather personal opinions than objective facts.
+This comparison will be limited to Python, Go, and Rust.
+Feel free to add more sections if you want to contrast TypeScript with another language.
+
+Some of the following points are based partially based on personal opinions.
```
Comment (roziscoding) [inline site/docs/resources/comparison.md:108]:
```suggestion
Some of the following points are partially based on personal opinions.
```

**Review by roziscoding — APPROVED** [review]
Minor suggestions on the english version :)

**Review by deptyped — APPROVED** [review]

File: site/docs/id/resources/comparison.md [hunk 12]
```diff
@@ -176,16 +129,35 @@ Karena beberapa orang memiliki selera yang berbeda, maka informasi-informasi ini
...
+
+Pilih grammY jika kamu merasa belum tahu ingin membangun sesuatu.
+TypeScript memungkinkanmu mengulangi basis kode dengan kecepatan luar biasa.
+Ia bagus untuk membuat prototipe dengan cepat, mencoba hal-hal baru, mengenal lebih tentang bot dan menyelesaikan kode dengan cepat. Sebagai aturan praktis, memproses ~100.000.000 pembaruan setiap hari dapat dilakukan dengan mudah menggunakan TypeScript, lebih dari itu akan memerlukan kerja tambahan, seperti menggunakan satu plugin grammY lagi.
+
+Pilih library yang ditulis dalam Go jika kamu sudah tahu dan sudah jelas apa yang ingin kamu buat (kamu juga tidak berharap membutuhkan banyak bantuan) dan kamu tahu kalau bot kamu akan memproses banyak hal.
+Sebagai bahasa yang di-compile secara native, Go lebih unggul dari TypeScript beberapa kali lipat dalam kecepatan asli CPU.
+Ini menjadi kurang relevan ketika kamu menulis bot yang kebanyakan waktunya habis untuk menunggu jaringan, padahal yang pentingnya itu seberapa cepat bot kamu mengurai JSON.
+Go bisa menjadi pilihan yang lebih baik dalam kasus ini.
```
Comment (quadratz) [inline site/docs/id/resources/comparison.md:152]:
```suggestion
Pilih library yang ditulis dalam Go jika kamu sudah tahu dan sudah jelas apa yang ingin kamu buat (tidak berharap membutuhkan banyak bantuan) dan tahu kalau bot kamu akan memproses banyak hal.
Sebagai bahasa yang di-compile secara native, Go lebih unggul dari TypeScript beberapa kali lipat dalam kecepatan pemrosesan CPU.
Keunggulan tersebut menjadi lebih relevan ketika kecepatan penguraian JSON menjadi hal yang krusial dan prioritas
Go bisa menjadi pilihan yang lebih baik dalam kasus ini.
```

File: site/docs/id/resources/comparison.md [hunk 13]
```diff
@@ -176,16 +129,35 @@ Karena beberapa orang memiliki selera yang berbeda, maka informasi-informasi ini
...
+
+### Framework yang Ditulis dalam Rust
+
+Hal serupa serupa [seperti pada Go](#framework-yang-ditulis-dalam-go), tetapi Rust lebih bagus.
+Kamu akan memerlukan lebih banyak waktu menulis dalam Rust, namun bot kamu juga akan lebih cepat tentunya.
+
+Mungkin menggunakan Rust itu menyenangkan, namun jarang keperluannya dalam membangun bot.
+Jika kamu ingin menggunakannya, maka gunakanlah, tapi pertimbangkanlah untuk mengatakan kalau kamu menyukai Rust hanya karena bahasanya dan Rust mungkin bukanlah alat yang tepat untuk pekerjaan tersebut.
```
Comment (quadratz) [inline site/docs/id/resources/comparison.md:160]:
```suggestion
### Framework yang Ditulis Menggunakan Rust

[Sama dengan Go](#framework-yang-ditulis-menggunakan-go), tetapi Rust lebih unggul.
Kamu akan memerlukan lebih banyak waktu menulis menggunakan Rust, namun bot kamu juga akan lebih cepat tentunya.

Meski menggunakan Rust itu menyenangkan, namun membangun bot menggunakan Rust terlalu berlebihan.
Jika kamu ingin menggunakannya, maka gunakanlah, tetapi jika kamu menyukai Rust hanya karena bahasanya, mungkin Rust bukanlah alat yang tepat untuk pekerjaan tersebut.
```

File: site/docs/id/resources/comparison.md [hunk 14]
```diff
@@ -50,83 +47,31 @@ Dengan begitu, kita bisa terhindar dari diskusi panas mengenai permasalahan typi
...
+  Telegraf tidak memilikinya---dokumentasinya diganti dengan sebuah referensi API yang kurang penjelasannya, beberapa panduan yang ada tidaklah lengkap dan sulit ditemukan;
+- grammY menggunakan TypeScript, type mereka _berfungsi dengan baik_ dan akan menyesuaikan diri dengan coding-an kamu.
   Di Telegraf kamu akan sering menulis kode dengan cara tertentu, yang jika tidak dilakukan, ia tidak akan bisa di-compile (meskipun tidak ada yang salah di kodenya); dan
 - grammY menambahkan beberapa petunjuk atau keterangan dari [referensi API Bot resmi](https://core.telegram.org/bots/api) yang akan membantu kamu ketika menulis kode.
   Telegraf sama sekali tidak menyediakan penjelasan di kode kamu.
+- Banyak hal lain seperti performa yang lebih baik, ekosistem plugin yang luas, dokumentasi yang telah diterjemahkan ke miliaran orang, integrasi dengan database dan framework web yang lebih baik, kompatibilitas runtime yang lebih baik, tersedianya [ekstensi VS Code](https://marketplace.visualstudio.com/items?itemName=grammyjs.grammyjs) dan beberapa hal lainnya yang nanti akan kamu temukan seiring berjalannya waktu.
```
Comment (dzikrl) [inline site/docs/id/resources/comparison.md:65]: @quadratz, how about this?

> integrasi dengan database dan framework web yang **lebih baik**, kompatibilitas runtime yang **lebih baik**

I'm running out of idea to reduce this repetition words.

Comment (dzikrl) [reply site/docs/id/resources/comparison.md:65]: I think it's okay.

File: site/docs/id/resources/comparison.md [hunk 15]
```diff
@@ -50,83 +47,31 @@ Dengan begitu, kita bisa terhindar dari diskusi panas mengenai permasalahan typi
...
-- Berjalan di Node.js, tetapi juga bisa berjalan di Deno dan browser.
+Berikut ada beberapa alasan mengapa kamu lebih baik menggunakan Telegraf daripada grammY.
 
-##### Kelebihan Telegraf
-
-- Lebih tua, berarti lebih berpengalaman.
-- Lebih banyak contoh bot yang tersedia, baik di repository mereka maupun di tempat lain.
+- Kamu telah mempunyai bot besar yang telah ditulis dalam Telegraf dan tidak ingin mengerjakannya lagi.
+  Dalam kasus tersebut, migrasi ke grammY akan memakan waktu yang lama daripada hasil untuk jangka panjangnya, tidak peduli walau migrasinya lancar.
+- Kamu mengenal Telegraf seperti punggung telapak tanganmu dan kamu tidak peduli untuk mengubah keahlianmu.
```
[the remark this answers is not in the document]
Comment (dzikrl) [reply site/docs/id/resources/comparison.md:71]: No, `seluk beluk` is the meaning. I just don't know how to figure it.

**Review by dzikrl — APPROVED** [review]

**Review by quadratz — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
Whoop whoop

**Pull request merged**